client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -0,0 +1,165 @@
+    loading,
+    error: queryError,
+    results,
+  } = useSearchQueries(options ?? {});
+
+  const debouncedOnChange = useDebounceCallback(
+    value => {
+      if (value.length >= MIN_CHARS) runQuery(value);
+      else {
+        if (results.length) runQuery('');
+      }
+    },
+    [searchText],
+    500
+  );
+
+  const handleDataUpdate = (selectedResult: Record<string, any> | null) => {
+    if (selectedResult === null) return;
+    if (!saveFields) handleChange(path, selectedResult);
+
+    const newObj: Record<string, any> = {};
+    saveFields?.forEach(
+      field => (newObj[field] = selectedResult[field] ?? null)
+    );
+    handleChange(path, newObj);
+    setEditMode(false);
+  };
+
+  if (!options) return null;
+
+  return (
+    <Box
+      display="flex"
+      alignItems="center"
+      gap={2}
+      style={{ minWidth: 300 }}
+      margin={0.5}
+      marginLeft={0}
+    >
+      <Box style={{ textAlign: 'end' }} flexBasis={FORM_LABEL_COLUMN_WIDTH}>
+        <FormLabel sx={{ fontWeight: 'bold' }}>{label}:</FormLabel>
+      </Box>
+      {editMode ? (
+        <Box flexBasis={FORM_INPUT_COLUMN_WIDTH} justifyContent="flex-start">
+          <>
+            <Autocomplete
+              sx={{ '.MuiFormControl-root': { minWidth: '100%' } }}
+              options={results}
+              disabled={!props.enabled}
+              onChange={(_, option) => handleDataUpdate(option)}
+              onInputChange={(_, value) => {
+                setSearchText(value);
+                debouncedOnChange(value);
+              }}
+              onBlur={() => {
+                if (data) setEditMode(false);
+              }}
+              getOptionLabel={getOptionLabel ?? undefined}
+              clearable={!props.config?.required}
+              inputProps={{
+                error: !!zErrors || !!props.errors || !!queryError,
+                helperText: zErrors ?? props.errors ?? queryError,
+              }}
+              noOptionsText={
+                loading
+                  ? t('control.search.searching-label')
+                  : t('control.search.no-results-label')
+              }
+              renderInput={params => (
+                <BasicTextInput
+                  {...params}
+                  placeholder={
+                    placeholderText ?? t('control.search.search-placeholder')
+                  }
+                />
+              )}
+            />
+          </>
+        </Box>
+      ) : (
+        <Box
+          display="flex"
+          alignItems="center"
+          justifyContent="space-between"
+          sx={{ width: FORM_INPUT_COLUMN_WIDTH }}
+        >
+          {getDisplayElement && getDisplayElement(data)}
+          <IconButton
+            label={t('label.edit')}
+            icon={<EditIcon style={{ width: 16 }} />}
+            onClick={() => {
+              setEditMode(true);
+            }}
+            color="primary"
+            height={'20px'}
+            width={'20px'}
```

Comment:
```suggestion
            height="20px"
            width="20px"
```

Reply:
Yeah, it's annoying how VSCode auto-prompts to that -- I normally catch them and fix them :) 

d4910a923